### PR TITLE
Add field ConflictsWith and Update CI, README

### PR DESCRIPTION
**Pull request by futuretea**
1. Add field ConflictsWith
2. Update README (move dev and test docs to https://github.com/harvester/terraform-provider-harvester/wiki)
3. Update CI to build and push master-head image
4. Add package script to build docker image

### CODE REVIEW

**Comment by futuretea** [comment]
## v0.3.0 Release note

Fixes
* Missing active network indication in UI when provisioning with Terraform [#1883](https://github.com/harvester/harvester/issues/1883)
* Now, If users specify ssh_keys in virtualmachine resource but didn't specify cloud-int userdata ssh_authorized_keys section, then will return an error to avoid user confusion [#1807](https://github.com/harvester/harvester/issues/1807)

Enhancements
* Support Data Sources in the Terraform Provider [#1838](https://github.com/harvester/harvester/issues/1838)

File: README.md [hunk 1]
```diff
@@ -9,12 +9,29 @@ Terraform Provider for Harvester
...
+provider "harvester" {
+  # Configuration options
+}
 ```
 
+see more info: https://registry.terraform.io/providers/harvester/harvester/latest
+
 ### Option 2: Build and Install The Provider Manually
 
+> :warning: The master version needs to be compiled and installed manually and can only be used for testing. Please use the latest release version in production
```
Comment (guangbochen) [inline README.md:33]: typo: append full stop to the end.

File: README.md [hunk 2]
```diff
@@ -44,15 +61,38 @@ The default location for locally-installed providers is one of the following, de
...
 terraform_harvester_provider_bin=./bin/terraform-provider-harvester
-
-terraform_harvester_provider_dir="${HOME}/.terraform.d/plugins/registry.terraform.io/harvester/harvester/${version}/${arch}/"
+terraform_harvester_providers_dir="${HOME}/.terraform.d/plugins/registry.terraform.io/harvester/harvester
+terraform_harvester_provider_dir="${terraform_harvester_providers_dir}/${version}/${arch}/"
 mkdir -p "${terraform_harvester_provider_dir}"
 cp ${terraform_harvester_provider_bin} "${terraform_harvester_provider_dir}/terraform-provider-harvester_v${version}"}
 ```
 
+After placing it into your plugins directory,  copy and paste this code into your Terraform configuration. Then, run `terraform init` to initialize it.
```
Comment (guangbochen) [inline README.md:73]: remove extra spaces before the `copy`.

File: README.md [hunk 3]
```diff
@@ -44,15 +61,38 @@ The default location for locally-installed providers is one of the following, de
...
+    }
+  }
+}
+
+provider "harvester" {
+  # Configuration options
+}
+```
+
+After testing, If you want to use the harvester provider from registry.terraform.io, you need to delete the local terraform provider harvester dir by:
```
Comment (guangbochen) [inline README.md:89]: If you want to switch back to the Harvester provider from `registry.terraform.io` after testing, you need to delete the local build manually via:

File: docs/index.md [hunk 4]
```diff
@@ -17,8 +17,9 @@ terraform {
   required_version = ">= 0.13"
   required_providers {
     harvester = {
-      source  = "registry.terraform.io/harvester/harvester"
-      version = "~> 0.2.9"
+      source = "harvester/harvester"
+      # Note: The master version needs to be compiled and installed manually and can only be used for testing. Please use the latest release version in production
+      version = "0.0.0-master"
```
[the remark this answers is not in the document]
Comment (futuretea) [reply docs/index.md:22]: will just keep using the latest version

**Comment by futuretea** [comment]
Updated

**Comment by futuretea** [comment]
```bash
docker run -itd --name terraform-provider-harvester-master-head futuretea/terraform-provider-harvester:master-head bash
docker exec -it terraform-provider-harvester-master-head bash
```
copy and paste this code into your Terraform configuration. Then, run `terraform init` to initialize it.
```hcl
terraform {
  required_providers {
    harvester = {
      source = "harvester/harvester"
      version = "0.0.0-master"
    }
  }
}

provider "harvester" {
  # Configuration options
}
```

File: .github/workflows/master.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+name: master
```
Comment (guangbochen) [inline .github/workflows/master.yml:1]: Unfortunately, we got to use the Drone CI only with docker credentials due to the infra team requirements.

**Review by guangbochen — APPROVED** [review]
LGTM

**Pull request merged**